src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/RequireThisCheck.java

```diff
@@ -1423,7 +1546,7 @@ public FrameType getType() {
     /**
      * A frame initiated on entering a for block; holds local for variable names.
      */
-    public static class ForFrame extends AbstractFrame {
+    private static class ForFrame extends AbstractFrame {
```

Comment:
Please move this change to separate commit and issue, we are library so we might not know all how we are used, some might subclass this Check. We need to declare such change.

But I am ok to make them private.

Reply:
There is six frame classes inside of `RequireThisCheck`: 
- [MethodFrame](https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/RequireThisCheck.java#L1129)
- [ConstructorFrame](https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/RequireThisCheck.java#L1150)
- [ClassFrame](https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/RequireThisCheck.java#1171)
- [AnonymousClassFrame](https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/RequireThisCheck.java#1359)
- [BlockFrame](https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/RequireThisCheck.java#1384)
- [CatchFrame](https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/RequireThisCheck.java#1405)
- [ForFrame](https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/RequireThisCheck.java#1426)

Olny `CatchFrame` #5351 and `ForFrame` #5392 are `public`.

@rnveach is there any reason to keep `CatchFrame` and `ForFrame` `public`?

Reply:
Not to my knowledge.
This may have been a mistake when I added them as I was also looking into creating a utility class like described at https://github.com/checkstyle/checkstyle/issues/5307 and probably copied it from that code when I made the PR. The classes had to be public in the util since they would be used by other classes.

Reply:
Ok, as it recent code, it is not on purpose.
But still better to move to separate commit, I am ok with minor prefix in commit message.

Reply:
@pbludov , please rebase

Reply:
done.